### ISSUE TITLE
Notion as RPM package

### ISSUE THREAD

**Issue by urbanware-org** [raboof/notion#225]
It would be great if you could provide Notion RPM packages again.

**Comment by raboof** [comment]
I agree it would be good if notion was easy to installed on RPM-based distro's.

Rather than providing RPM's directly from the notion project, though, wouldn't it make more sense to make sure Notion makes it into the official repo's?

In the past this might have been slightly problematic because we were under the custom 'ion' license, but from notion4 onwards we are on the familiar lgpl again, so that should make this easier.

**Comment by wilhelmy** [comment]
You can find the .spec file used by OpenSuSE [here](https://build.opensuse.org/package/show/X11%3Awindowmanagers/notion), I have packaged for OpenSuSE before with medium success so I could help with that platform at least. I guess packages for other RPM based systems can be made conformant with the packaging style of whichever distro it is you want to submit it to. I'll try to make time for more notion stuff around christmas.

**Comment by urbanware-org** [comment]
> Rather than providing RPM's directly from the notion project, though, wouldn't it make more sense to make sure Notion makes it into the official repo's?

Indeed! That would be even better, of course.

> I'll try to make time for more notion stuff around christmas.

Thanks for your efforts!

**Comment by wilhelmy** [comment]
This should work but I haven't tried building or using it on fedora or basically any other rpm based distro that is not OpenSuSE. https://build.opensuse.org/request/show/789142

I think maybe it's possible to build notion for those on the OpenSuSE build system as well, but I haven't tried (and I assume the dependency packages would have different names). Seems to take them a while to merge my request, too.

Which distro do you use? I'm relatively sure the sane way to distribute something that links external libraries like X11 and lua and cairo and so on would be to submit a package to your distribution.

**Comment by urbanware-org** [comment]
Thanks for the information!

Currently, I use Fedora 31 on my systems. As soon as I have the time (on the upcoming weekend I guess) I will check it out and get back to you.

**Comment by urbanware-org** [comment]
Sorry for the delay, it was a quite busy time...

Anyway, I tried to install the OpenSuSE rpm package which failed because of dependency errors:
```
# rpm -i notion-4.0.0-8.1.x86_64.rpm
warning: notion-4.0.0-8.1.x86_64.rpm: Header V3 DSA/SHA1 Signature, key ID bcc2c97d: NOKEY
error: Failed dependencies:
    liblua.so.5.2()(64bit) is needed by notion-4.0.0-8.1.x86_64
    libreadline.so.6()(64bit) is needed by notion-4.0.0-8.1.x86_64
#
```
I checked for `liblua` first. The `/usr/lib64` directory contains version 5.1 and 5.3 of the library, but version 5.2 is missing. I tried to "fix" that using a symlink for version 5.2 which simply points on version 5.3 of the library, but without success (still says that the library is missing).

The same applies to `libreadline` (version 6 is required, version 8 is installed).

**Comment by wilhelmy** [comment]
You need to rebuild the package for your distribution.

**Comment by urbanware-org** [comment]
Ah, I see. So far, I haven't (re-)built a package myself, but there seem to be plenty how-tos. As soon as I have some spare time, I'll proceed with that.

I thought the library symlink fiddling was worth a try as it worked in some cases in the past, even though it's not the best solution.

**Comment by wilhelmy** [comment]
You can probably reuse parts of the specfile for opensuse that I've linked above. If you need help, maybe ask on the fedora IRC. I have no idea how things work there.

**Comment by urbanware-org** [comment]
I have got some news about this.

There is an **unofficial** repository which provides RPM packages from current versions:

```
sudo dnf copr enable ecook/notion
sudo dnf install notion
```

Just installed the latest version via package manager and it works fine.

Maybe there is a way to get the packages into the base repositories. I will get in touch with the provider of the package.

Anyway, I assume we can close this issue now?

**Comment by wilhelmy** [comment]
Thanks, I think it would be nice to have this in the documentation.

**Comment by raboof** [comment]
since the repo is unofficial, perhaps we should keep this issue open so it can be found as a workaround, but we should not document/close it until it has been accepted into the official repo's?

**Comment by urbanware-org** [comment]
That all sounds good. I would also go along with that decision.
